File: asset-transfer-basic/chaincode-typescript/src/asset.ts
/*
 * SPDX-License-Identifier: Apache-2.0
 */
import { Object, Property } from 'fabric-contract-api';

@Object()

export class Asset {
    @Property()
    public docType?: string;
    public ID: string;
    public Color: string;
    public Size: number;
    public Owner: string;
    public AppraisedValue: number;
}

File: full-stack-asset-transfer-guide/contracts/asset-transfer-go/smartcontract/asset.go
/*
SPDX-License-Identifier: Apache-2.0
*/

package smartcontract

// Asset represents a tradeable asset in the world state.
type Asset struct {
	ID             string `json:"ID"`
	Color          string `json:"Color"`
	Owner          string `json:"Owner"`
	AppraisedValue int    `json:"AppraisedValue"`
	Size           int    `json:"Size"`
}
